cmd/export_ledgers.go

```diff
@@ -54,13 +82,56 @@ func printTransformStats(attempts, failures int, printLog bool) {
 
 	results, err := json.Marshal(resultsMap)
 	if err != nil {
-		cmdLogger.Fatal("Could not marshall results: ", err)
+		cmdLogger.Fatal("Could not marshal results: ", err)
+	}
+
+	cmdLogger.Info(string(results))
+}
+
+func generateRunId() string {
+	return uuid.New().String()
+}
+
+func uploadToGcs(credentialsPath, bucket, runId, path string) error {
+	// Use credentials file in dev/local runs. Otherwise, derive credentials from the service account.
+	if len(credentialsPath) > 0 {
+		os.Setenv("GOOGLE_APPLICATION_CREDENTIALS", credentialsPath)
+		cmdLogger.Infof("Using credentials found at: %s", credentialsPath)
+	}
+
+	reader, err := os.Open(path)
+	if err != nil {
+		return fmt.Errorf("failed to open file %s: %v", path, err)
+	}
+
+	ctx := context.Background()
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to create client: %v", err)
+	}
+	defer client.Close()
+
+	ctx, cancel := context.WithTimeout(ctx, time.Second*60)
+	defer cancel()
+
+	objectPath := filepath.Join(runId, path)
+	wc := client.Bucket(bucket).Object(objectPath).NewWriter(ctx)
+	defer wc.Close()
+
+	cmdLogger.Infof("Uploading %s to gs://%s/%s ", path, bucket, objectPath)
+
+	if _, err := io.Copy(wc, reader); err != nil {
```

Comment:
Are there any limitations to file size that can be uploaded? Sometimes with historical loading the `operations` cmd utility can generate 1-2GB files.

Reply:
Seems like the limit for one object is [5 TB](https://cloud.google.com/storage-transfer/docs/known-limitations-transfer), so 1-2GB should theoretically be ok.